packages/block-library/src/search/edit.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * WordPress dependencies
+ */
+import { Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { RichText, InspectorControls } from '@wordpress/editor';
+import { PanelBody, TextControl } from '@wordpress/components';
+
+export default function SearchEdit( { attributes, setAttributes } ) {
+	const { label, placeholder } = attributes;
+
+	return (
+		<Fragment>
+			<div className="wp-block-search">
+				<RichText
+					tagName="div"
+					className="wp-block-search__label"
+					value={ label }
+					formattingControls={ [] }
+					onChange={ ( text ) => setAttributes( { label: text } ) }
+				/>
+				<input
```

Comment:
@melchoyce or @mapk - how would you feel about using inline editing for the placeholder? We would have to style `RichText` to look like an input field but it might be closer to the principles that blocks approach promotes.

Reply:
I think we should give it a try — going to be hard to judge until we can test it in-context.